publication-rest/src/main/java/no/unit/nva/publication/PublicationResponseFactory.java

```diff
@@ -0,0 +1,72 @@
+package no.unit.nva.publication;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.util.Optional;
+import java.util.Set;
+import no.unit.nva.PublicationMapper;
+import no.unit.nva.api.PublicationResponse;
+import no.unit.nva.api.PublicationResponseElevatedUser;
+import no.unit.nva.clients.IdentityServiceClient;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifact;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifactList;
+import no.unit.nva.model.associatedartifacts.file.HiddenFile;
+import no.unit.nva.publication.permission.strategy.PublicationPermissionStrategy;
+import nva.commons.apigateway.RequestInfo;
+
+public final class PublicationResponseFactory {
+
+    private PublicationResponseFactory() {
+    }
+
+    public static PublicationResponse create(Publication publication, RequestInfo requestInfo,
+                                             IdentityServiceClient identityServiceClient) {
+        var userStrategy = getPublicationPermissionStrategy(requestInfo, publication, identityServiceClient);
```

Comment:
can this be `.map(this:: hasAuthenticatedAccessOnPublication).orElse(false)`?

Reply:
forgot hasAuthenticatedAccessOnPublication, maybe

        return userStrategy.filter(PublicationResponseFactory::hasAuthenticatedAccessOnPublication)
                   .map(strategy -> (PublicationResponse) createAuthenticatedResponse(strategy, publication))
                   .orElse(createPublicResponse(publication));